### PR TITLE
Homework5.1

### PR DESCRIPTION
**Pull request by Borisko130**

### CODE REVIEW

File: src/main/java/core/basesyntax/Main.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Main {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Main.java:3]: we don't need main in PR, let's remove it

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
 .idea/*
 *.iml
 target/*
+src/main/java/core/basesyntax/Main.java
```
Comment (boroda4436) [inline .gitignore:4]: Bad practice.

File: src/main/java/core/basesyntax/Engine.java [hunk 3]
```diff
@@ -1,5 +1,22 @@
...
+    Engine(int volume, int numberOfCylinders, int fuelConsumption, int horsePower, int weight) {
+        this.volume = volume;
+        this.numberOfCylinders = numberOfCylinders;
+        this.fuelConsumption = fuelConsumption;
+        this.horsePower = horsePower;
+        this.weight = weight;
+    }
+
+    @Override
+    protected Engine clone() throws CloneNotSupportedException {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Engine.java:19]: does it really throw that exception or it's has a grey color?

File: src/main/java/core/basesyntax/Car.java [hunk 4]
```diff
@@ -1,5 +1,32 @@
...
+        this.model = model;
+    }
+
+    public void describe() {
+        System.out.println(model + " " + maxSpeed + " "
+                + amountOfDoors + " " + isTruck + " " + engine);
+    }
+
+    @Override
+    protected Car clone() throws CloneNotSupportedException {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:29]: same

File: src/main/java/core/basesyntax/Car.java [hunk 5]
```diff
@@ -1,5 +1,32 @@
...
-public class Car {
+public class Car implements Cloneable {
 
+    private String model;
+    private int maxSpeed;
+    private int amountOfDoors;
+    private boolean isTruck;
+    private Engine engine;
+
+    Car(String model, int maxSpeed, int amountOfDoors, boolean isTruck, Engine engine) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:11]: what is the default access modifier this constructor will have?

File: src/main/java/core/basesyntax/Engine.java [hunk 6]
```diff
@@ -1,5 +1,22 @@
...
 
-public class Engine {
+public class Engine implements Cloneable {
+    private int volume;
+    private int numberOfCylinders;
+    private int fuelConsumption;
+    private int horsePower;
+    private int weight;
 
+    Engine(int volume, int numberOfCylinders, int fuelConsumption, int horsePower, int weight) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Engine.java:10]: same

**Pull request closed without merging**